statistics/cmsketch.go

```diff
@@ -339,6 +361,13 @@ func DecodeCMSketch(data []byte) (*CMSketch, error) {
 	if len(p.Rows) == 0 {
 		return nil, nil
 	}
+	// Sometimes, topNData mightcontains older top n elements when count of new top n is less than old top n.
+	if len(topNData) < len(p.TopN) {
+		return nil, errors.Trace(errors.New("length of topNData and p.TopN mismatch"))
```

Comment:
Shall we use `topNData` when mismatch happens and avoid raising error?

Reply:
No, if mismatch happens, one of cmsketch and topn must be wrong, we should notice the user and they should call fast analyze again to avoid wrong query plan.

Reply:
Then shall we check `len(topNData) != len(p.TopN)` instead?

Reply:
Yeah, it is used to deal with dirty data that not overwritten by REPLACE INTO before, but now there should not have any dirty data.

Reply:
Since we are using delete + insert now, we don't need this check anymore?